src/global-config/network-selection.ts

```diff
@@ -42,35 +42,31 @@ export function useNetworkSelector() {
 
   const selectedNetworkQueryParam = searchParams.get("network") ?? "";
 
-  function selectNetwork(
-    network: NetworkName,
-    {replace = false}: {replace?: boolean} = {},
-  ) {
-    if (!isValidNetworkName(network)) return;
-    setSearchParams(
-      (prev) => {
-        const newParams = new URLSearchParams(prev);
-        newParams.set("network", network);
-        return newParams;
-      },
-      {replace},
-    );
-    writeSelectedNetworkToLocalStorage(network);
-  }
-
-  // on init check for existence of network query param, if not present, check local storage for a previously selected network. Then set query param to the network defined in local storage.
-  useEffect(
-    () => {
-      const currentNetworkSearchParam = searchParams.get("network");
-      if (!isValidNetworkName(currentNetworkSearchParam ?? "")) {
-        selectNetwork(getUserSelectedNetworkFromLocalStorageWithDefault(), {
-          replace: true,
-        });
-      }
+  const selectNetwork = useCallback(
+    (network: NetworkName, {replace = false}: {replace?: boolean} = {}) => {
+      if (!isValidNetworkName(network)) return;
+      setSearchParams(
+        (prev) => {
+          const newParams = new URLSearchParams(prev);
+          newParams.set("network", network);
+          return newParams;
+        },
+        {replace},
+      );
+      writeSelectedNetworkToLocalStorage(network);
     },
-    [], // empty [] makes this effect only run once (on mount)
```

Comment:
I think this is a case where we most likely want the previous behaviour and only run this effect on initial mount. Perhaps add some `// eslint-ignore <rule-name>` here to stop the linter from complaining about it.

Let's just undo the code change in this file and add this eslint ignore comment